src/sarif_converter.ts

```diff
@@ -96,61 +96,65 @@ const getTools = (analysisResults: IAnalysisResult, suggestions: ISarifSuggestio
     rules.push(rule);
 
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    result[suggestionIndex] = {
-      ...suggestions[suggestionIndex],
-      ruleIndex,
-      rule,
-      level: severity,
-      id: suggestionId,
-      text: suggestion.message,
-    };
+    const allIssuesOfSuggestion: ISarifSuggestion[] = allIssuesBySuggestions[suggestionIndex];
+    allIssuesOfSuggestion.forEach(issue => {
```

Comment:
NIT: I would vote more for `.map()` without side effects,
```
allIssues.push(...allIssuesOfSuggestion.map(issue => {...})
```

Reply:
There are multiple changes I would like to do as part of this file - the code is really complex....

Reply:
As I'm short in time, and this is an important change, I would like us to do more improvements like that soon .... maybe next sprint?